Adyen/UI/List/ListSection.swift

```diff
@@ -25,17 +30,39 @@ public struct ListSection {
     ///   - header: The section header.
     ///   - items: The items inside the section.
     ///   - footer: The section footer.
-    public init(header: ListSectionHeader? = nil, items: [ListItem], footer: ListSectionFooter? = nil) {
+    public init(header: ListSectionHeader? = nil,
+                items: [ListItem],
+                footer: ListSectionFooter? = nil) {
         self.header = header
         self.items = items
         self.footer = footer
+        self.identifier = UUID().uuidString
+    }
+    
+    private let identifier: String
+    
+    internal mutating func deleteItem(index: Int) {
+        guard items.indices.contains(index) else { return }
+        items.remove(at: index)
+    }
+    
+    public func hash(into hasher: inout Hasher) {
+        hasher.combine(identifier)
```

Comment:
the identifier is when header and footer are nil, but the identifier alone is not enough, this is safer, in case for example we change it so that the header is changed to being mutable and someone change the header after creation. so right now both approaches are identical, but by adding the header and footer its more future proof. and more accurate tbh.

using only the identifier is kind of a hack.